hack/observability/alloy/kustomization.yaml

```diff
@@ -2,12 +2,12 @@ resources:
   - ../namespace.yaml
 
 helmCharts:
-  - name: promtail
+  - name: alloy
     repo: https://grafana.github.io/helm-charts
-    releaseName: promtail
+    releaseName: alloy
     namespace: observability
     valuesFile: values.yaml
-    version: 6.16.6
+    version: 0.12.3
 
 helmGlobals:
     # Store chart in ".charts" folder instead of "charts".
```

Comment:
We need to add the following at the end to create the configmap which contaings the data of config.alloy:

```yaml
namespace: observability

configMapGenerator:
- name: alloy-config
  files:
  - config.alloy

```